experimental/compactdecimal/src/compactdecimal.rs

```diff
@@ -306,15 +314,15 @@ impl CompactDecimalFormatter {
     /// #     });
     /// #
     /// let about_a_million = CompactDecimal::from_str("1.20c6").unwrap();
-    /// let three_million = CompactDecimal::from_str("3c6").unwrap();
+    /// let three_million = CompactDecimal::from_str("+3c6").unwrap();
     /// let ten_lakhs = CompactDecimal::from_str("10c5").unwrap();
     /// # // The following line contains U+00A0 NO-BREAK SPACE.
     /// assert_writeable_eq!(short_french.format_compact_decimal(&about_a_million).unwrap(), "1,20 M");
     /// assert_writeable_eq!(long_french.format_compact_decimal(&about_a_million).unwrap(), "1,20 million");
     ///
     /// # // The following line contains U+00A0 NO-BREAK SPACE.
-    /// assert_writeable_eq!(short_french.format_compact_decimal(&three_million).unwrap(), "3 M");
-    /// assert_writeable_eq!(long_french.format_compact_decimal(&three_million).unwrap(), "3 millions");
+    /// assert_writeable_eq!(short_french.format_compact_decimal(&three_million).unwrap(), "+3 M");
+    /// assert_writeable_eq!(long_french.format_compact_decimal(&three_million).unwrap(), "+3 millions");
```

Comment:
Yes, but try this in a locale where the sign and the compact affix are on the same side. For example:

https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-numbers-full/main/sw-KE/numbers.json

We have a pattern

```
              "1000000-count-other": "M0",
```

The correct output according to CLDR would be "+M3", but your code produces "M+3"

Reply:
(moved to the other thread instead of starting a new thread)